Tests/SwiftFusionTests/Inference/GaussianFactorGraphTests.swift

```diff
@@ -0,0 +1,48 @@
+import SwiftFusion
+import TensorFlow
+import XCTest
+
+final class GaussianFactorGraphTests: XCTestCase {
+  
+  func Vector2_t(_ x: Double, _ y: Double) -> Tensor<Double> {
+    let t = Tensor<Double>(shape: [2, 1], scalars: [x, y])
+    return t
+  }
+  
+  /// Factor graph with 2 2D factors on 3 2D variables
+  func createSimpleGaussianFactorGraph() -> GaussianFactorGraph {
+    var fg = GaussianFactorGraph()
+    
+    let I_2x2: Tensor<Double> = eye(rowCount: 2)
+    let x1 = 2, x2 = 0, l1 = 1;
```

Comment:
Can we have non-int keys as well?

Reply:
I am thinking about that, but let's first focus on making this all work :)

Reply:
One way to do this is to make a type-erased container that is `Equatable`